crates/leon/src/lib.rs

```diff
@@ -59,7 +59,7 @@
 //! # let template = Template::parse("hello {name}").unwrap();
 //! assert_eq!(
 //!     template.render(
-//!         &vals(|_key| Some("marcus".into()))
+//!         &&vals(|_key| Some("marcus".into()))
```

Comment:
The && is unfortunate but ah well, worth it

Reply:
> The && is unfortunate but ah well, worth it

Yeah with this `&&` I was able to remove the lifetime annotation in `leon::Values`, accept non-`'static` `Fn` and return non-`'static` values.

Before that, the `Fn` must be `'static`, so it can either return a `&'static str` or `String`, not very useful.